lib/Raven/Middleware/ContextInterfaceMiddleware.php

```diff
@@ -51,17 +58,14 @@ public function __construct(Context $context)
      */
     public function __invoke(Event $event, callable $next, ServerRequestInterface $request = null, \Exception $exception = null, array $payload = [])
     {
-        $tagsContext = isset($payload['tags_context']) ? $payload['tags_context'] : [];
-        $extraContext = isset($payload['extra_context']) ? $payload['extra_context'] : [];
-        $serverOsContext = isset($payload['server_os_context']) ? $payload['server_os_context'] : [];
-        $runtimeContext = isset($payload['runtime_context']) ? $payload['runtime_context'] : [];
-        $userContext = isset($payload['user_context']) ? $payload['user_context'] : [];
+        if (isset($payload[$this->contextName . '_context'])) {
+            $method = preg_replace_callback('/_[a-zA-Z]/', function ($matches) {
+                return strtoupper($matches[0][1]);
+            }, 'with_' . $this->contextName . '_context');
```

Comment:
You should add a method existence check here, to detect errors

Reply:
Also, why underscores and not camelCase?

Reply:
>You should add a method existence check here, to detect errors

Done

>Also, why underscores and not camelCase?

The context aliases (the names that are used both in the `$payload` argument of the middleware and in the event data that is sent) are in underscore notation while the methods of the `Event` class are in camel case. In this case I transform the whole method name in camel case with a single regex

Reply:
>You should add a method existence check here, to detect errors

Done

>Also, why underscores and not camelCase?

The context aliases (the names that are used both in the `$payload` argument of the middleware and in the event data that is sent) are in underscore notation while the methods of the `Event` class are in camel case. In this case I transform the whole method name to camel case with a single regex

Reply:
Then I find this fuction highly unreadable, and poor performance-wise. I would prefer a dumb, easy-to-read, lightning fast switch case, since the number of context names are low and well known: you even have constants to use!

Reply:
>Then I find this fuction highly unreadable

Indeed, I focused too much on making something too generic where there is was no real need